### PR TITLE
interp: fix reading from external global

### PR DESCRIPTION
**Pull request by aykevl**
This fixes https://github.com/tinygo-org/tinygo/issues/3020.

### CODE REVIEW

**Review by dgryski — APPROVED** [review]
LGTM

**Comment by deadprogram** [comment]
Thank you for the fix @aykevl and for the review @dgryski merging!

File: interp/interpreter.go [hunk 1]
```diff
@@ -369,6 +369,11 @@ func (r *runner) run(fn *function, params []value, parentMem *memoryView, indent
 				nBytes := uint32(operands[3].Uint())
 				dstObj := mem.getWritable(dst.index())
 				dstBuf := dstObj.buffer.asRawValue(r)
+				if mem.get(src.index()).buffer == nil {
+					// Looks like the source buffer is not defined.
+					// This can happen with //extern or //go:embed.
+					return nil, mem, r.errorAt(inst, errUnsupportedRuntimeInst)
```
Comment (codefromthecrypt) [inline interp/interpreter.go:375]: for those watching at home, this error is recoverable (interp.isRecoverableError). returning this error doesn't fail, it just defers evaluation until runtime (this line is executed compiling the code).

Comment (dkegel-fastly) [reply interp/interpreter.go:375]: thanks for the explanation!

**Pull request merged**